base/commands/multimap/multimap_lock.go

```diff
@@ -0,0 +1,53 @@
+//go:build base || multimap
+
+package _multimap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec"
+)
+
+type MultiMapLockCommand struct{}
+
+func (m MultiMapLockCommand) Init(cc plug.InitContext) error {
+	addKeyTypeFlag(cc)
+	cc.AddIntFlag(multiMapTTL, "", ttlUnset, false, "time-to-live (ms)")
+	help := "Lock a key in the given MultiMap"
+	cc.SetCommandHelp(help, help)
+	cc.SetCommandUsage("lock [key] [flags]")
+	cc.SetPositionalArgCount(1, 1)
+	return nil
+}
+
+func (m MultiMapLockCommand) Exec(ctx context.Context, ec plug.ExecContext) error {
+	mmName := ec.Props().GetString(multiMapFlagName)
+	ttl := GetTTL(ec)
+	ci, err := ec.ClientInternal(ctx)
+	if err != nil {
+		return err
+	}
+	keyStr := ec.Args()[0]
+	keyData, err := makeKeyData(ec, ci, keyStr)
+	if err != nil {
+		return err
+	}
+	req := codec.EncodeMultiMapLockRequest(mmName, keyData, 0, ttl, 0)
```

Comment:
Do we need to expose the TTL for lock operations? Go client implementation does not support ttl for lock operations. Maybe we could just use client implementation since we are not encoding/decoding any data?

Reply:
I saw that client has `lockWithLease` which supports the TTL but it also has support for the lock context. So it might be better to just use that function.